tests/integration/test_wazuh_db/data/agent_messages.yaml

```diff
@@ -1,77 +1,207 @@
-    stage: "agent vuln_cve clearing table again"
+    stage: "agent vuln_cves missing payload"
+  -
+    input: 'agent 000 vuln_cves insert {"name":"test_package2",
+                                        "version":"3.0",
+                                        "architecture":"x86",
+                                        "cve":"CVE-2021-1001",
+                                        "reference":"99efe684b5ff4646b3c754de46cb6a9cbee9fbaa",
+                                        "type":"PACKAGE",
+                                        "status":"VALID",
+                                        "check_pkg_existence":false}'
+    output: 'ok {"action":"INSERT","status":"SUCCESS"}'
+    stage: "agent vuln_cves insert package with same CVE without checking if the package is present in sys_programs"
+  -
+    input: 'agent 000 sql SELECT * FROM vuln_cves WHERE name = "test_package2"'
+    output: 'ok [{"name":"test_package2","version":"3.0","architecture":"x86","cve":"CVE-2021-1001","reference":"99efe684b5ff4646b3c754de46cb6a9cbee9fbaa","type":"PACKAGE","status":"VALID"}]'
+    stage: "agent vuln_cves checking package insertion with same CVE"
+  -
+    input: 'agent 000 vuln_cves insert {"name":"test_package2",
+                                        "version":"3.0",
+                                        "architecture":"x86",
+                                        "cve":"CVE-2021-1002",
+                                        "reference":"99efe684b5ff4646b3c754de46cb6a9cbee9fbaa",
+                                        "type":"PACKAGE",
+                                        "status":"VALID",
+                                        "check_pkg_existence":false}'
+    output: 'ok {"action":"INSERT","status":"SUCCESS"}'
+    stage: "agent vuln_cves insert same package with different CVE without checking if the package is present in sys_program"
+  -
+    input: 'agent 000 sql SELECT * FROM vuln_cves WHERE name = "test_package2" AND cve = "CVE-2021-1002"'
+    output: 'ok [{"name":"test_package2","version":"3.0","architecture":"x86","cve":"CVE-2021-1002","reference":"99efe684b5ff4646b3c754de46cb6a9cbee9fbaa","type":"PACKAGE","status":"VALID"}]'
+    stage: "agent vuln_cves checking package with different CVE"
+  -
+    input: 'agent 000 vuln_cves update_status {"old_status":"PENDING",
+                                               "new_status":"OBSOLETE"}'
+    output: 'ok'
+    stage: "agent vuln_cves update specific status to another one"
+  -
+    input: 'agent 000 sql SELECT distinct status FROM vuln_cves'
+    output: 'ok [{"status":"OBSOLETE"},{"status":"VALID"}]'
+    stage: 'agent vuln_cves checking change specific status by another one'
+  -
+    input: 'agent 000 vuln_cves update_status {"new_status":"PENDING",
+                                               "type":"OS"}'
+    output: 'ok'
+    stage: "agent vuln_cves update status by type"
+  -
+    input: 'agent 000 sql SELECT distinct status FROM vuln_cves'
```

Comment:
This DISTINCT statement confirms that at least one vulnerability has changed to PENDING, but it doesn't confirm that all OS type vulnerabilities have changed.

Reply:
Yes, I fixed it using count the same as I did later to at least have a number of how many rows were updated.